### PR TITLE
[tflite] enable passing int32 elementray arith to NNAPI

### PR DESCRIPTION
**Pull request by freedomtan**
NNAPI 1.3 (Android API level 30) [1] supports int32 add, sub, mul, and div. This patch enables passing +-*/ with int32 tensors to NNAPI. With that, we can do something like
```
./mul_test   --gtest_filter=IntegerMulOpTest.NoActivation --use_nnapi=1
```
on Android R devices, such as Pixel 4 running Android R beta.

[1] https://developer.android.com/ndk/reference/group/neural-networks

### CODE REVIEW

**Review by miaowang14 — DISMISSED** [review]
Thanks for your work!

**Review by rthadur — DISMISSED** [review]

**Comment by rthadur** [comment]
@freedomtan this changes are breaking this test https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/add_test.cc#L208 , can you please check ?

File: tensorflow/lite/delegates/nnapi/nnapi_delegate.cc [hunk 1]
```diff
@@ -1553,7 +1615,11 @@ bool NNAPIDelegateKernel::Validate(
   switch (builtin_code) {
     case kTfLiteBuiltinAdd: {
       ExpectMaxOpVersion(version, 2, &val_ctx);
-      ExpectIsFloatOrQuant8Operator(context, node, &val_ctx);
+      if (android_sdk_version >= kMinSdkVersionForNNAPI13) {
```
Comment (miaowang14) [inline tensorflow/lite/delegates/nnapi/nnapi_delegate.cc:1618]: NNAPI in Android R requires the fused activation function to be NONE:
https://android.googlesource.com/platform/frameworks/ml/+/refs/heads/master/nn/runtime/include/NeuralNetworks.h#263

Similarly for MUL, SUB, DIV.

Comment (freedomtan) [reply tensorflow/lite/delegates/nnapi/nnapi_delegate.cc:1617]: changes added. Thanks.

File: tensorflow/lite/delegates/nnapi/nnapi_delegate.cc [hunk 2]
```diff
@@ -1968,9 +2040,16 @@ bool NNAPIDelegateKernel::Validate(
       ExpectOpVersion(version, 1, &val_ctx);
       ExpectMinAndroidSdkVersion(android_sdk_version, kMinSdkVersionForNNAPI11,
                                  &val_ctx);
-      Expect(context->tensors[node->inputs->data[0]].type == kTfLiteFloat32,
-             NNAPIValidationFailureType::kUnsupportedInputType,
-             "NNAPI only support float div.", &val_ctx);
+      const TfLiteType input_type =
+          context->tensors[node->inputs->data[0]].type;
+
+      if (android_sdk_version >= kMinSdkVersionForNNAPI13) {
```
[the remark this answers is not in the document]
Comment (freedomtan) [reply tensorflow/lite/delegates/nnapi/nnapi_delegate.cc]: reverted DIV part, thanks.

**Comment by freedomtan** [comment]
@miaowang14 and @rthadur: I reverted DIV part and added activation function check, and ran add, sub, mul kernel tests with `--with_nnapi=1`

**Review by miaowang14 — APPROVED** [review]

**Review by miaowang14 — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Pull request merged**